src/libs/pluginprocess/pluginprocess.c

```diff
@@ -396,26 +470,71 @@ int elektraPluginClose (Plugin * handle, Key * errorKey)
  * @retval 0 if the data structure is still used
  * @ingroup processplugin
  **/
-int elektraPluginProcessClose (ElektraPluginProcess * pp, Key * errorKey)
+ElektraPluginProcessCloseResult elektraPluginProcessClose (ElektraPluginProcess * pp, Key * errorKey)
 {
 	pp->counter = pp->counter - 1;
+	int result = elektraPluginProcessSend (pp, ELEKTRA_PLUGINPROCESS_CLOSE, NULL, errorKey);
 	int done = pp->counter <= 0;
 	if (done) cleanupPluginData (pp, errorKey);
-	return done;
+	ElektraPluginProcessCloseResult closeResult = { result, done };
+	return closeResult;
 }
 
 /** Store a pointer to any plugin related data that is being executed inside an own process.
+ *
+ * This is required in case additional arbitrary plugin data should be stored. Pluginprocess
+ * has to be stored using elektraPluginSetData. Plugin data for the child process
+ * has to be stored using this function like
+ * @code
+int elektraPluginOpen (Plugin * handle, Key * errorKey)
+{
+	ElektraPluginProcess * pp = elektraPluginGetData (handle);
+	if (pp == NULL)
+	{
+		if ((pp = elektraPluginProcessInit (errorKey)) == NULL) return ELEKTRA_PLUGIN_STATUS_ERROR;
+		ArbitraryPluginData * data = // initialize your plugin data
+		elektraPluginProcessSetData (handle, data);
+		elektraPluginSetData (handle, pp);
+		if (!elektraPluginProcessIsParent (pp)) elektraPluginProcessStart (handle, pp);
+	}
+	if (elektraPluginProcessIsParent (pp)) return elektraPluginProcessOpen (pp, errorKey);
+
+	// actual plugin functionality to be executed in a child process
+	return ELEKTRA_PLUGIN_STATUS_SUCCESS;
+}
+ * @endcode
+ *
+ * Furthermore ensure to cleanup the data after the plugin is done like
+ * @code
+int elektraPluginClose (Plugin * handle, Key * errorKey)
+{
+	ElektraPluginProcess * pp = elektraPluginGetData (handle);
+	if (elektraPluginProcessIsParent (pp)) {
+		ArbitraryPluginData * data = elektraPluginProcessGetData (handle);
+		ElektraPluginProcessCloseResult result = elektraPluginProcessClose (pp, errorKey);
+		if (result.cleanedUp)
+		{
+			elektraPluginSetData (handle, NULL);
+			// cleanup your plugin data here
+		}
```

Comment:
else? Or what do we need the cleanedUp for?

Reply:
cleanedUp indicates when the whole process is over and thus ElektraPluginProcess data got cleared already, it is then time to cleanup any plugin-related data if there is some. The else action is simply doing nothing

Reply:
Maybe rename cleanedUp to processFinished?

But isn't this wrong if the "close" call did not clean up?